### PR TITLE
[PKG-5559] Initial build 1.2.2 ❄️ 

### PR DESCRIPTION
**Pull request by ViridianMelody**
libsndfile 1.2.2 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-5559]
- [Upstream repository](https://github.com/libsndfile/libsndfile/tree/1.2.2)
- [Upstream changelog/diff](https://github.com/libsndfile/libsndfile/blob/1.2.2/CHANGELOG.md)

### Explanation of changes:

- Skipping Windows and s390x for missing dependencies (libflac, lame, mpg123, libopus)
- Removed patch connected to issue [fixed in 1.2.0](https://github.com/libsndfile/libsndfile/pull/828).
- Added mpg123 to run as test command failed without it.

[PKG-5559]: https://anaconda.atlassian.net/browse/PKG-5559?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,59 +8,50 @@ package:
...
-    - libopus
-    - lame
+    - libflac 1.4.3
+    - libvorbis 1.3.7
+    - libogg 1.3.5
+    - libopus 1.3
+    - lame 3.100
+    - mpg123 1.30.0
+  run:
     - mpg123
```
Comment (cbouss) [inline recipe/meta.yaml]: Should use pin_compatible here. mpg123 does not have run_exports.

**Review by Arishamays1 — APPROVED** [review]

**Comment by Arishamays1** [comment]
LGTM after Charles' suggestion

File: recipe/meta.yaml [hunk 2]
```diff
@@ -8,59 +8,50 @@ package:
...
-    - lame
-    - mpg123
+    - libflac 1.4.3
+    - libvorbis 1.3.7
+    - libogg 1.3.5
+    - libopus 1.3
+    - lame 3.100
+    - mpg123 1.30.0
+  run:
+    - {{ pin_compatible('mpg123') }}
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: We probably want to be more strict than that. 1.25.8 has `lib/libmpg123.so.0.44.7` while 1.30.0 has `lib/libmpg123.so.0.47.0`. So I don't know how they version their DSO files, but they don't necessarily seem to follow a logical pattern.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: More strict as in, we probably want to put an upper bound on the minor version just to be safe.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: Though, https://abi-laboratory.pro/?view=timeline&l=mpg123 is quit clean... Maybe an upper bound on the major is enough...

**Review by boldorider4 — APPROVED** [review]

**Pull request merged**